backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/MigrateVmCommand.java

```diff
@@ -429,10 +428,18 @@ private MigrateVDSCommandParameters createMigrateVDSCommandParameters() {
         if (parallelMigrations == null) {
             maxIncomingMigrations = maxOutgoingMigrations = effectiveMigrationPolicy.getMaxMigrations();
         }
-        if (getVm().getCpuPinningPolicy() == CpuPinningPolicy.DEDICATED) {
-            cpuSets = CpuPinningHelper.getAllPinnedPCpus(getDedicatedCpuPinning(getDestinationVdsManager())).stream()
-                    .map(Object::toString).collect(Collectors.toList());
-            String numaPinningString = vmHandler.createNumaPinningForDedicated(getVm(), getDestinationVdsId());
+        if (getVm().getCpuPinningPolicy().isExclusive()) {
+            String cpuPinning = getExclusiveCpuPinning(getDestinationVdsManager());
+            List<CpuPinningHelper.PinnedCpu> pinnedCpus = CpuPinningHelper.parseCpuPinning(cpuPinning);
+            cpuSets = new LinkedList<>();
+            for (CpuPinningHelper.PinnedCpu pin : pinnedCpus) {
+                var pCpus = pin.getpCpus().stream().collect(Collectors.toList());
+                if (!pCpus.isEmpty()) {
+                    cpuSets.add(pCpus.get(0).toString());
+                }
```

Comment:
this code seems a bit awkward now - can we have a PinnedCpu without pinning? and why do we need to collect the pcpus to list if we just take the first element there?..

Reply:
Generally speaking, `PinnedCpu` pinning can't be null, but in terms of code it's possible to be an empty list. I prefer being on the safe side and check if it's empty.
I did the collect to switch it from collection to a list. I think we do care about the order, so better switch it to a list.

Reply:
> Generally speaking, `PinnedCpu` pinning can't be null, but in terms of code it's possible to be an empty list. I prefer being on the safe side and check if it's empty.

ok..

> I did the collect to switch it from collection to a list. I think we do care about the order, so better switch it to a list.

when would pCpus.get(0) be different than pin.getpCpus().get(0)?

Reply:
Collection<Integer> pCpus doesn't have get().